cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -264,16 +264,15 @@ else()
                  ${ARROW_BOOST_BUILD_VERSION})
   set_urls(
     BOOST_SOURCE_URL
+    "https://dl.bintray.com/boostorg/release/${ARROW_BOOST_BUILD_VERSION}/source/boost_${ARROW_BOOST_BUILD_VERSION_UNDERSCORES}.tar.gz"
```

Comment:
Please prefer the trimmed boost bundles over full boost bundles

Reply:
You mean "https://dl.bintray.com/ursalabs/arrow-boost/boost_${ARROW_BOOST_BUILD_VERSION_UNDERSCORES}.tar.gz" ?

Reply:
Yes, please see the code comment that explains what it is. See also ARROW-8222.